### DISCUSSION TITLE
D365 Deprecations

### DISCUSSION THREAD

**Discussion by imartinflores** [XRM-OSS/D365-UI-Test#41] [General]
Hello, I think  you might be aware of this, but just in case, there will be some deprecations related to the Xrm object, hence, some changes might be  needed:

https://docs.microsoft.com/en-us/power-platform/important-changes-coming#deprecation-of-office365-authentication-type-and-organizationserviceproxy-class-for-connecting-to-dataverse

**Comment by todstanis** [comment]
Has anyone investigated the impact of the deprecations? 

I manage a QA group using Playwright with C# to write automation tests for our Dynamics apps. While I don't use this library directly, I've taken some of the ideas that allow tests to get around dependency on the DOM and used them from C# through the Evaluate methods. 

If the Xrm.Page is truly going away, it seems that will have a significant impact on this library. Reading into the changes as well as reading any questions posed by other developers in other forums, I've found no good/correct answer on how to get the executionContext outside of a method called from a Dynamics customization. Unless I'm missing something, it would appear this library will not work in the future, as there's currently no way gain access to the methods provided by Xrm.Page, without that executionContext, and there is no way this or any similar library can get that executionContext, that I'm aware.

Hopefully I'm just missing something. If anyone knows how to get the executionContext or formContext directly, I'd love to know.

**Comment by DigitalFlow** [comment]
Hi everyone,

for the time being it's really uncertain what will happen and the deprecation has changed back and forth quite a few times.
So for now, I'm watiting to see what MS really does.

There was an undocumented workaround that was posted a while ago: https://community.dynamics.com/365/f/dynamics-365-general-forum/299666/xrm-page-deprecation-and-a-new-way-of-getting-the-execution-context-from-an-html-web-resource

I'm not too fond of it, but if Xrm.Page gets removed, we might give it a try.
As a last resort there's always the possibility of writing a short Dynamics Library that can be registered on all Entity Forms which should be UI tested and to just export the executionContext to a global variable.

So whatever happens, we'll find a way to make it work. 

Kind regards,
Florian
